src/MonitoringFunctions.Common/HelperMethods.cs

```diff
@@ -26,21 +28,35 @@ internal static class HelperMethods
         /// <param name="monitorName">Name of this monitor to be included in the logs and in the data sent to Kusto.</param>
         /// <param name="url">Url that this method will attempt to access.</param>
         /// <returns>A task, tracking the initiated async operation. Errors should be reported through exceptions.</returns>
-        internal static async Task CheckAndReportUrlAccessAsync(ILogger log, string monitorName, string url, IDataService dataService,
+        internal static async Task CheckAndReportUrlAccessAsync(ILogger log,
+            string monitorName,
+            string url,
+            IDataService dataService,
             CancellationToken cancellationToken = default)
         {
-            HttpResponseMessage response = await _httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
-
-            await dataService.ReportUrlAccessAsync(monitorName, response, cancellationToken).ConfigureAwait(false);
-
-            if (response.IsSuccessStatusCode)
+            _ = log ?? throw new ArgumentNullException(paramName: nameof(log));
+            _ = string.IsNullOrWhiteSpace(monitorName) ? throw new ArgumentNullException(paramName: nameof(monitorName)) : monitorName;
+            _ = string.IsNullOrWhiteSpace(url) ? throw new ArgumentNullException(paramName: nameof(url)) : url;
+            _ = dataService ?? throw new ArgumentNullException(paramName: nameof(dataService));
+            try
             {
-                log.LogInformation($"Monitor '{monitorName}' has succeeded accessing the url {url}");
+                using HttpResponseMessage response = await _httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
+                HttpRequestLogEntry entry = CreateHttpRequestLogEntry(monitorName, response.StatusCode, response.RequestMessage.RequestUri.AbsoluteUri);
+                await dataService.ReportUrlAccessAsync(entry, cancellationToken).ConfigureAwait(false);
+                if (response.IsSuccessStatusCode)
+                {
+                    log.LogInformation($"Monitor '{monitorName}' succeeded accessing url {url}.");
+                    return;
+                }
+                string error = $"Monitor '{monitorName}' failed accessing url {url} with error code {response.StatusCode}.";
+                log.LogError(error);
+                throw new Exception(error);
```

Comment:
Would it be beneficial to use specific Exception class for this error rather than general System.Exception?

Reply:
According to [docs](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getasync?view=net-5.0#System_Net_Http_HttpClient_GetAsync_System_String_) different versions of the framework return different exceptions. It seems to be safer to just catch `Exception` here.

Reply:
I was talking about `throw new Exception(error);` but seems this code is gone